src/switchroot/ostree-prepare-root.c

```diff
@@ -394,13 +392,13 @@ main (int argc, char *argv[])
       // https://github.com/systemd/systemd/blob/604b2001081adcbd64ee1fbe7de7a6d77c5209fe/src/basic/mountpoint-util.h#L36
       // which bumps up these defaults for the rootfs a bit.
       g_autofree char *root_upperdir
-          = root_transient ? g_build_filename (OTCORE_RUN_OSTREE_PRIVATE, "root/upper", NULL)
+          = rootfs_config->root_transient ? g_build_filename (OTCORE_RUN_OSTREE_PRIVATE, "root/upper", NULL)
                            : NULL;
       g_autofree char *root_workdir
-          = root_transient ? g_build_filename (OTCORE_RUN_OSTREE_PRIVATE, "root/work", NULL) : NULL;
+          = rootfs_config->root_transient ? g_build_filename (OTCORE_RUN_OSTREE_PRIVATE, "root/work", NULL) : NULL;
```

Comment:
This scares me... I am not sure if it's the correct way to do this since we are calling into rootfs_config without a if statement before.... 

@cgwalters

Reply:
You mean handling if `rootfs_config` is `NULL`? I think that can't happen; we dereference the pointer earlier too, but most importantly we checked for `NULL` on line 287

But yeah, this is where Rust removes all this kind of uncertainty about NULL pointers...

Reply:
yeah that exactly was my fear, thanks!